docs/devel/networking.md

```diff
@@ -72,6 +72,74 @@ As of now, the existent binding mechanisms are:
 - [slirp](#slirp-binding-mechanism)
 
 ### Bridge binding mechanism
+Using the bridge `BindMechanism` requires a VMI configuration featuring a
+network whose interface type is `bridge` - the yaml file below can be used
+as reference, but please refer to the
+[user guide](https://kubevirt.io/user-guide/#/creation/interfaces-and-networks?id=bridge)
+for more information.
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+        - name: default
+          bridge: {}
+  networks:
+  - name: default
+    pod: {} # Stock pod network
+```
+
+Let's refer to the image below to get a better understanding of how this
+`BindMechanism` works.
+
+Bridge binding mechanism diagram
+
+![alt text](https://github.com/kubevirt/kubevirt.github.io/blob/source/assets/images/diagram.png)
+
+As can be seen in the diagram above, there are three actors at play: CNI,
+libvirt, and DHCP. For completeness sake, let's add one more actor that is
+implicit in the picture: KubeVirt.
+
+Once the bridge `BindMechanism` is triggered,
+[CNI](https://github.com/containernetworking/plugins/tree/master/plugins/main/bridge)
```

Comment:
It is dangerous to mention the bridge CNI in context of bridge binding. They are used together, but the fact that they are both `bridge` is just a coincidence. People may get an impression that `bridge` binding is for `bridge` CNI.

I think that the document should split the two stages of network configuration - Pod network and VM network (binding). Then you can explain that there are CNI plugins (the default Kubernetes network, additional CNIs, ...) and that there are methods to extend the network they provide to the VM.

Reply:
Aye aye captain, good point.

I hope the distinction is clearer now.

> Except the inline comments, as I mentioned in the parent PR, IMO the links to the source could be avoided and moved to a blog instead of here.
> Although they do help me a lot now, I think with time they will loose sync with this document.

You did make really good questions which made me realize I was not telling the whole story. Thanks for "forcing" me get to the bottom of this.